### PR TITLE
Bugfix/orders

### PR DESCRIPTION
**Pull request by eineving**
Orders now functioning due to change to String Dates instead of Date Object

### CODE REVIEW

File: backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java [hunk 1]
```diff
@@ -73,6 +88,22 @@ public Order updateOrder(String orderID, Order order) {
         return getOrder(orderID);
     }
 
+    /**
+     * Dates should be on format yyyy-MM-dd
+     *
+     * @param date
+     */
+    private void validateDate(String date) throws InvalidDateFormatException {
```
Comment (chraker) [inline backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java]: MINOR Remove the declaration of thrown exception 'se.chalmers.dat265.group1.api.order.InvalidDateFormatException' which is a runtime exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java [hunk 2]
```diff
@@ -73,6 +88,22 @@ public Order updateOrder(String orderID, Order order) {
...
+        //TODO
```
Comment (chraker) [inline backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

File: backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java [hunk 3]
```diff
@@ -5,6 +5,10 @@
 import se.chalmers.dat265.group1.model.dbEntities.OrderData;
 import se.chalmers.dat265.group1.model.dbEntities.OrderedPart;
 
+import java.text.DateFormat;
```
Comment (chraker) [inline backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java]: MINOR Remove this unused import 'java.text.DateFormat'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java [hunk 4]
```diff
@@ -36,34 +42,43 @@ public Order getOrder(String orderID) {
...
-            orderedPart.setID(id);
+            orderedPart.setOrderID(id);
             orderedPartRepository.postObject(orderedPart);
         }
         return getOrder(id + "");
     }
 
     @Override
-    public Order updateOrder(String orderID, Order order) {
+    public Order updateOrder(String orderID, Order order) throws InvalidDateFormatException {
```
Comment (chraker) [inline backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java]: MINOR Remove the declaration of thrown exception 'se.chalmers.dat265.group1.api.order.InvalidDateFormatException' which is a runtime exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java [hunk 5]
```diff
@@ -5,6 +5,10 @@
 import se.chalmers.dat265.group1.model.dbEntities.OrderData;
 import se.chalmers.dat265.group1.model.dbEntities.OrderedPart;
 
+import java.text.DateFormat;
+import java.text.ParseException;
```
Comment (chraker) [inline backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java]: MINOR Remove this unused import 'java.text.ParseException'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java [hunk 6]
```diff
@@ -36,34 +42,43 @@ public Order getOrder(String orderID) {
     }
 
     @Override
-    public Order createNewOrder(Order order) {
+    public Order createNewOrder(Order order) throws InvalidDateFormatException {
```
Comment (chraker) [inline backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java]: MINOR Remove the declaration of thrown exception 'se.chalmers.dat265.group1.api.order.InvalidDateFormatException' which is a runtime exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderAPI.java [hunk 7]
```diff
@@ -13,7 +13,7 @@
 
     Order getOrder(String orderID);
 
-    Order createNewOrder(Order order);
+    Order createNewOrder(Order order) throws InvalidDateFormatException;
```
Comment (chraker) [inline backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderAPI.java]: MINOR Remove the declaration of thrown exception 'se.chalmers.dat265.group1.api.order.InvalidDateFormatException' which is a runtime exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderAPI.java [hunk 8]
```diff
@@ -13,7 +13,7 @@
...
 
-    Order updateOrder(String orderID, Order order);
+    Order updateOrder(String orderID, Order order) throws InvalidDateFormatException;
```
Comment (chraker) [inline backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderAPI.java]: MINOR Remove the declaration of thrown exception 'se.chalmers.dat265.group1.api.order.InvalidDateFormatException' which is a runtime exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by chraker** [comment]
SonarQube analysis reported 8 issues
* ![MINOR][MINOR] 7 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by JBerlin84 — APPROVED** [review]

**Pull request merged**